backend/src/main/java/com/bakdata/conquery/models/query/DistributedExecutionManager.java

```
@@ -127,18 +149,36 @@ public ManagedExecution createQuery(QueryDescription query, UUID queryId, User u
 	/**
 	 * Receive part of query result and store into query.
 	 *
-	 * @param result
+	 * @implNote subQueries of Forms are managed by the form itself, so need to be passed from outside.
 	 */
-	public <R extends ShardResult, E extends ManagedExecution & InternalExecution<R>> void handleQueryResult(R result) {
 
-		final ManagedExecutionId executionId = result.getQueryId();
-		final E query = (E) storage.getExecution(executionId);
+	@SneakyThrows
+	public <R extends ShardResult, E extends ManagedExecution & InternalExecution<R>> void handleQueryResult(R result, E query) {
+
+
+		log.debug("Received Result[size={}] for Query[{}]", result.getResults().size(), result.getQueryId());
+		log.trace("Received Result\n{}", result.getResults());
 
 		if (query.getState() != ExecutionState.RUNNING) {
 			return;
```

Oh hier silently droppen wir doch ein ShardResult, oder wird vorher schon etwas geloggt?